src/NLog/Internal/StringHelpers.cs

```diff
@@ -51,10 +51,7 @@ public static class StringHelpers
         internal static bool IsNullOrWhiteSpace(string value)
         {
 #if NET3_5
-
-            if (value == null) return true;
-            if (value.Length == 0) return true;
-            return String.IsNullOrEmpty(value.Trim());
+            return value?.Length > 0 ? string.IsNullOrEmpty(value.Trim()) : true;
```

Comment:
Do you find this really easier to read?

Reply:
Yes I do, and the compiler also think it is better. But I can guess from your leading question, that I should change it back?

Reply:
Hoe is this better for the compiler? I'm pretty sure that an equal comparison is better than a larger than in term of performance

Reply:
Less instructions are usually better for the compiler, but I will revert. Since this is not important.